src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
+  const arrayFromDate = date.split(oldSeparator);
+  const result = [];
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    for (let j = 0; j < fromFormat.length; j++) {
+      if (toFormat[i] === 'YYYY' && fromFormat[j] === 'YY') {
+        fromFormat[j] = 'YYYY';
+
+        if (arrayFromDate[j] < 30) {
+          arrayFromDate[j] = '20' + arrayFromDate[j];
+        } else {
+          arrayFromDate[j] = '19' + arrayFromDate[j];
+        }
+      }
+
+      if (toFormat[i] === 'YY' && fromFormat[j] === 'YYYY') {
+        fromFormat[j] = 'YY';
+
+        const newFormatYear = arrayFromDate[j].slice(2);
+
+        arrayFromDate[j] = newFormatYear;
+      }
+
+      if (toFormat[i] === fromFormat[j]) {
+        result[i] = arrayFromDate[j];
+      }
```

Comment:
The condition `if (toFormat[i] === fromFormat[j])` assumes that the formats will match directly. However, this logic does not account for cases where the formats need conversion, such as between 'YY' and 'YYYY'. Ensure that all necessary conversions are handled before this check.